openerp/addons/base/ir/ir_qweb.py

```diff
@@ -59,7 +59,7 @@ def raise_qweb_exception(etype=None, **kw):
             e.qweb[k] = v
         # Will use `raise foo from bar` in python 3 and rename cause to __cause__
         e.qweb['cause'] = original
-        raise
+        raise etype(e.qweb.items)
 
```

Comment:
AFAIK doing this will lose the original traceback.
IMO `raise` alone is right, but QWebException maybe deserves a custom `__repr__` that shows more information about the qweb values.